### DISCUSSION TITLE
Get progress bar for a Python GUI

### DISCUSSION THREAD

**Discussion by dridk** [duckdb/duckdb#7100] [Q&A] [answered]
Hi,

I would like to show the duckdb progress  bar in my python GUI . Is it possible to get it ? 

Something like this would be greate : 

```
conn = duckdb.connect("...)
conn.progress_changed(callback )



```

**Comment by Tishj** [comment]
I'm not sure I follow, you want to register a callback that's invoked every time the progress changed?

Or do you want to manually update the progress bar?

**Reply by dridk** [reply]
The first one, to update my gui progress bar.
SQLite do as follow : 
https://docs.python.org/3/library/sqlite3.html?highlight=progress%20sqlite#sqlite3.Connection.set_progress_handler

**Reply by Tishj** [reply, marked as answer]
I have a working implementation on my fork, needs some proper testing and dependency management work though (to keep the python objects alive)

I added `set_progress_handler` to the connection, which takes a callback function + a state object, to pass to the callback

The callback function should take: the state, a float for the progress, a bool to indicate finished

Would that be sufficient?

**Reply by dridk** [reply]
Yes it is perfect !
